modules/post/multi/manage/shell_to_meterpreter.rb

```diff
@@ -94,8 +94,10 @@ def run
         payload_name = 'windows/meterpreter/reverse_tcp'
         psh_arch = 'x86'
       else
-        print_error('Target is running Windows on an unsupported architecture such as Windows ARM!')
-        return nil
+        unless datastore['PAYLOAD_OVERRIDE']
```

Comment:
I might be wrong, but I don't think this logic is right 👀 

I assumed if the user set `PAYLOAD_OVERRIDE`, then the `session.platform` detection would be skipped entirely? I wonder if you were only testing this with a windows machine maybe 👀